include/realizations/catchment/Bmi_C_Adapter.hpp

```diff
@@ -29,22 +24,30 @@ namespace models {
 
         public:
 
-            explicit Bmi_C_Adapter(std::string forcing_file_path, bool model_uses_forcing_file, bool allow_exceed_end,
+            explicit Bmi_C_Adapter(std::string library_file_path, std::string forcing_file_path,
+                                   bool model_uses_forcing_file, bool allow_exceed_end,
                                    bool has_fixed_time_step, utils::StreamHandler output);
 
-            Bmi_C_Adapter(std::string bmi_init_config, std::string forcing_file_path, bool model_uses_forcing_file,
+            Bmi_C_Adapter(std::string library_file_path, std::string bmi_init_config,
+                          std::string forcing_file_path, bool model_uses_forcing_file,
                           bool allow_exceed_end, bool has_fixed_time_step, utils::StreamHandler output);
 
-            Bmi_C_Adapter(std::string forcing_file_path, bool model_uses_forcing_file, bool allow_exceed_end,
-                          bool has_fixed_time_step, const geojson::JSONProperty& other_input_vars,
+            Bmi_C_Adapter(std::string library_file_path, std::string forcing_file_path,
+                          bool model_uses_forcing_file, bool allow_exceed_end,
+                          bool has_fixed_time_step, const geojson::JSONProperty &other_input_vars,
                           utils::StreamHandler output);
 
-            Bmi_C_Adapter(const std::string& bmi_init_config, std::string forcing_file_path,
+            Bmi_C_Adapter(std::string library_file_path, const std::string &bmi_init_config,
+                          std::string forcing_file_path,
                           bool model_uses_forcing_file, bool allow_exceed_end, bool has_fixed_time_step,
-                          const geojson::JSONProperty& other_input_vars, utils::StreamHandler output);
+                          const geojson::JSONProperty &other_input_vars, utils::StreamHandler output);
 
             // Copy constructor
-            Bmi_C_Adapter(Bmi_C_Adapter &adapter);
+            // TODO: since the dynamically loaded lib and model struct can't easily be copied (without risking breaking
```

Comment:
Making a copy of the handle does seem problematic, and even if it is POSSIBLE using serialize/deserialize, I don't know that copying is a semantic that would be useful.  I'm thinking that maybe just move semantics is really what is wanted, so that all BMI formulations are singleton by nature.

Reply:
Yeah, I don't think we could actually copy it.  We'd have to something more complicated; e.g., create a more complicated construct around holding it, and use a copyable shared pointer to that type.  Regardless, I agree it's something not worth worrying about unless/until something actually compels us to.